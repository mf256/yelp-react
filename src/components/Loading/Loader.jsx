import React from 'react';
import styled from 'styled-components';
import GridLoader from 'react-spinners/GridLoader';
import { mainColor } from '../GlobalStyles/GlobalStyles';

const StyledLoader = styled.div`
  margin: 4rem auto;
  width: 57px;
`;

class Loader extends React.PureComponent {
  render() {
    return (
      <StyledLoader>
        <GridLoader
          color={mainColor}
          loading
        />
      </StyledLoader>
    );
  }
}

export default Loader;
