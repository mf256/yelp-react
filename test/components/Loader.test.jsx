import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../../src/components/Loading/Loader';

describe('Loader', () => {
  it('should match snapshot', () => {
    const component = shallow(<Loader />);

    expect(component).toMatchSnapshot();
  });
});
