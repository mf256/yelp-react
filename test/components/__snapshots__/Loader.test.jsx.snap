// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
<styled.div>
  <Loader
    color="#fa7d19"
    css=""
    loading={true}
    margin={2}
    size={15}
  />
</styled.div>
`;
